### PR TITLE
📣 No more celery for starting flows and sending broadcasts

### PR DESCRIPTION
**Pull request by rowanseymour**
* Sends broadcasts by creating a mailroom task
* Uses a simplified send function to continue sending broadcasts in tests
* Simplify `Flow.start` to no longer use `start_msg_flow_batch`
* Removes `start_msg_flow_batch` celery task (flows queue should be empty now)
* Moves code only used by legacy engine to `temba.msgs.legacy`

### CODE REVIEW

File: temba/mailroom/queue.py [hunk 1]
```diff
@@ -66,6 +63,23 @@ def queue_mailroom_mo_miss_task(event):
     queue_mailroom_contact_task(event.org_id, event.contact_id, MO_MISS_EVENT_TASK, event_task)
 
 
+def queue_mailroom_broadcast_task(broadcast):
+    """
+    Queues the passed in broadcast for sending by mailroom
+    """
+
+    task = {
+        "translations": dict(broadcast.text),
```
Comment (nicpottier) [inline temba/mailroom/queue.py]: need `template_state` = `legacy` and translations needs to be a dict of lang to `{"text":""}`

File: temba/msgs/tests.py [hunk 2]
```diff
@@ -841,21 +841,6 @@ def test_flows(self):
         self.assertEqual(set(response.context["object_list"]), {msg3, msg2, msg1})
         self.assertEqual(response.context["actions"], ["label"])
 
-    def test_footgun(self):
```
Comment (rowanseymour) [inline temba/msgs/tests.py]: it's on mailroom to test this now

File: temba/utils/queues.py [hunk 3]
```diff
@@ -15,7 +15,6 @@
 # celery queue names
 class Queue:
     CELERY = "celery"
-    FLOWS = "flows"
```
Comment (rowanseymour) [inline temba/utils/queues.py]: we should be able to completely remove "flows" (and "msgs") queues after this pr

File: temba/channels/models.py [hunk 4]
```diff
@@ -28,7 +28,7 @@
 from django.utils.http import urlquote_plus
 from django.utils.translation import ugettext_lazy as _
 
-from temba.mailroom.queue import queue_mailroom_mo_miss_task
+from temba import mailroom
```
Comment (rowanseymour) [inline temba/channels/models.py]: just cleaning up how we access mailroom functionality so it's always through the `mailroom` module

File: temba/flows/models.py [hunk 5]
```diff
@@ -1968,25 +1830,18 @@ def start_msg_flow_batch(
                 entry_rules.flow = self
 
         msgs_to_send = []
-        optimize_sending_action = len(broadcasts) > 0
```
Comment (rowanseymour) [inline temba/flows/models.py]: never bother with this optimization now that this code is only used for testing

File: temba/mailroom/queue.py [hunk 6]
```diff
@@ -63,23 +63,67 @@ def queue_mailroom_mo_miss_task(event):
...
+        "flow_id": start.flow_id,
+        "flow_type": start.flow.flow_type,
+        "contact_ids": list(start.contacts.values_list("id", flat=True)),
+        "group_ids": list(start.groups.values_list("id", flat=True)),
+        "restart_participants": start.restart_participants,
+        "include_active": start.include_active,
+        "extra": start.extra,
+    }
+
+    _queue_batch_task(org_id, START_FLOW_TASK, task, HIGH_PRIORITY)
```
Comment (rowanseymour) [inline temba/mailroom/queue.py]: HIGH_PRIORITY was used previously when this code was in `FlowStart.start_in_mailroom`.. but is it correct @nicpottier ?

Comment (nicpottier) [reply temba/mailroom/queue.py]: Yep, that's right. The batches will be created at high priority but each batch will be at normal priority.

Comment (rowanseymour) [reply temba/mailroom/queue.py]: So should https://github.com/nyaruka/rapidpro/blob/master/temba/mailroom/queue.py#L86 be HIGH_PRIORITY as well ?

**Pull request merged**